sopel/bot.py

```diff
@@ -376,47 +377,6 @@ def reply(self, text, dest, reply_to, notice=False):
         else:
             self.say(text, dest)
 
-    class SopelWrapper(object):
```

Comment:
I'm on board with this. Adding yet another sub-namespace in the form of `sopel.wrapper` seems like overkill.

Reply:
Done. Resolved the resulting circular dependency: 
> from sopel.bot import SopelWrapper leads to import errors because sopel.bot wants sopel.module which wants sopel.test_tools, which will want SopelWrapper from sopel.bot...

in a less-than-ideal way (violates "convention" of imports at the top; this file already defers one import: `import sys`, so...) by moving `import sopel.test_tools` into `module.example.__call__` where it was needed. 

Let me know if instead this issue should be resolved with a better refactoring of the current test functions between `test_tools.py` and `module.example`.

Reply:
Yeah, that makes sense for now. It's not ideal, but I consider this PR to be a first step toward a cleaner interface for both production and testing purposes.

So go for it, just add a TODO comment. :)